js/client-chat.js

```diff
@@ -1022,8 +1036,11 @@
 				}
 			}
 			if (userlist) this.addRow(userlist);
-			if (autoscroll) {
+			if (this.getAutoScroll()) {
 				this.$chatFrame.scrollTop(this.$chat.height());
+			} else if (!this.scrollHandler) {
+				this.scrollHandler = _.bind(ConsoleRoom.scrollHandler, this);
+				this.$chatFrame.on('scroll', this.scrollHandler);
```

Comment:
The rest makes sense to me, but this is kind of confusing. Why are you attaching an extra scroll handler here?